cmodel.c

```diff
@@ -20,17 +20,24 @@ bool reset = 0;
 void checkInterrupt(void);
 
 enum {
+  PORT_JOYBUS_WRITE = 0,
+  PORT_JOYBUS_READ = 2,
```

Comment:
Are you sure that port 2 is the Joybus Read port? The SM5K6 datasheet says `inl` reads from port 3, and port 2 already seems to be occupied (for resetting Joybus channel, and some other signal?)

Reply:
it seems so given the usage. Have a look at `joybusTransferChannel` where it's pretty clear that it's used symmetrically to `PORT_JOYBUS_WRITE`.